.github/workflows/publish-major-minor.yml

```
@@ -0,0 +1,51 @@
+name: "Publish Major-Minor-Tags"
+on:
+  push:
+    tags:
+      - "v*"
+
+jobs:
+  push-tags:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install packaging
+        run: python -m pip install -U packaging --user
+      - name: Get versions
+        id: get_versions
+        shell: python
+        run: |
+          from packaging.version import parse
+
+          tag_ref = "${{ github.ref }}"
+          tag_name = tag_ref.split("/")[-1]
+          version = parse(tag_name)
+          print(f"tag_name: {tag_name}")
+          print(f"version: {version}")
+          if not version.is_prerelease:
+              print("Creating new major and minor tags!")
+              print(f"::set-output name=original_tag_name::{tag_name}")
+              print(f"::set-output name=major_version::v{version.major}")
+              print(f"::set-output name=minor_version::v{version.major}.{version.minor}")
+          else:
+              print("No tags created (dev or pre version)!")
```

I still prefer a guard expression here to reduce nesting
```suggestion
          if not version.is_prerelease:
              print("No tags created (dev or pre version)!")
              return

          print("Creating new major and minor tags!")
          print(f"::set-output name=original_tag_name::{tag_name}")
          print(f"::set-output name=major_version::v{version.major}")
          print(f"::set-output name=minor_version::v{version.major}.{version.minor}")
```

---

@s-weigand you know why it's preferable to accept suggestions via GH UI and then pull locally, right?

---

@webknjaz I'm not sure I understand what you mean 😕 
Is it about this 'Commit suggestions' button, I just realized that it's there?

Normally I use the suggested changes as a kind of TODO list, first reading all to check if issues are overlapping and giving a 👍 if it is a no-brainer to implement it. When I start implementing the changes I start out with those suggestions and resolve the conversations step by step to keep track of what still needs to be done. And when I'm done implementing all changes I don't need further feedback for, I push the commits to the feature branch so the changes can be reevaluated.

---

Since the code isn't wrapped in a function I think the if part should look like this:
```python
if version.is_prerelease:
    print("No tags created (dev or pre version)!")
    exit(0)
```

---

Pro tip: if you use the `Commit suggestions` button, it preserves the authorship of the suggestion submitter meaning that you can credit changes to all the collaborators. It'll add `Co-Authored-By` trailers to the commit message and so even if you pull and decide to polish things, or squash commits, you'll be able to preserve all of the authors. Also, you can apply suggestions as batches if they are related.

---

Ohhh that's nice, thanks for sharing, I will try that later 😄